src/components/atoms/Checkbox/Checkbox.scss

```diff
@@ -0,0 +1,108 @@
+@import "scss/constants";
+
+$checkbox-diameter: 32px;
+
+$checkbox-transition: all 200ms cubic-bezier(0.23, 1, 0.32, 1);
+$checkbox-shadow-preset: inset 0 0 0 1px;
+
+$toggler-width: 56px;
+$toggler-height: 30px;
+
+$toggler-circle-diameter: 26px;
+$toggler-circle-spacing: 2px;
+
+$toggler-circle-moving-delta: $toggler-width - $toggler-circle-diameter -
+  $toggler-circle-spacing;
+
+.Checkbox {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+
+  position: relative;
+
+  cursor: pointer;
+
+  // unsetting global label styles
+  margin: 0;
+
+  &__native-input {
+    &:checked + .Checkbox__custom-input {
+      background-color: $primary;
+
+      .Checkbox__tick-icon {
+        opacity: 1;
+        transform: scale(1);
+      }
+
+      &--toggler {
+        background-color: $dimmer-primary;
+
+        &::after {
+          left: $toggler-circle-moving-delta;
+
+          background-color: $primary;
+        }
+      }
+    }
+  }
```

Comment:
Yes, they react to the `native state` of that hidden native input. These styles change the appearance of checkbox, when the state is `checked = true`

Reply:
Fair enough, makes sense. Would it be better to use `~` (sibling) rather than `+` (adjacent sibling) for the selector to make it less brittle in case the structure of the code is changed?

- https://www.w3schools.com/css/css_combinators.asp

Reply:
Though.. I wonder.. given it's dealing with 'siblings', can these style rules be added to the non-hidden things, while still reacting to the changes on the hidden native element? It would seem like that should still work, and would be less confusing IMO.

Reply:
>  Would it be better to use ~ (sibling)

Done in https://github.com/sparkletown/sparkle/pull/1299/commits/fa746a0d247b163eae4a4127a1228bd96131c4a2

> It would seem like that should still work, and would be less confusing IMO.

I don't think there is a selector for that. The `~` will not work in this case

![image](https://user-images.githubusercontent.com/63194656/117037491-34fc5980-ad0f-11eb-9d3f-72aa4b074641.png)

Resolving this, if you feel that there is a way to do that, unresolve

Reply:
As i've requested in the past, please don't just screenshot docs without providing a link to them as that provides no context for me to read further/validate/etc.

It seems like you were probably using:

- https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator

---

Reading that, it seems that it won't work as I originally suggested (the wording in the docs I was basing it off (linked above) implied that it was all siblings, not all siblings that followed afterwards)

Reply:
Though.. thinking about it.. would you not just be able to solve that by moving the hidden native element after the current 'visible div/etc' stuff to allow the classes to be fixed up?

Reply:
I don't understand what is your request and how does moving the hidden input can help us

Reply:
You said that the ~ can only target siblings that come after the element, and that that was a blocker on refactoring the BEM classes as I originally asked about.

So if the hidden input was moved after, then the classes could be rearranged as the ~ selector would let us target things as we want to then.